doc/man3/CMS_add0_cert.pod

```diff
@@ -22,22 +22,20 @@ CMS_add0_crl, CMS_add1_crl, CMS_get1_crls
 
 CMS_add0_cert() and CMS_add1_cert() add certificate I<cert> to I<cms>
 unless it is already present.
-This is used by L<CMS_sign_ex(3)> and L<CMS_sign(3)> and may be used before
-calling L<CMS_verify(3)> to help chain building in certificate validation.
+This can be used with L<CMS_sign_ex(3)> and L<CMS_sign(3)>
+to include a chain certificate or an extra signer certificate
+and may be used with L<CMS_verify(3)> to set a
+fallback signer certificate or to help chain building in certificate validation.
```

Comment:
This is packs too many cases into too little text. The original is IMHO better by not trying to say as much.

A more clear description might be (if I am not mistaken): The certificates "added" to the CMS structure are available to the verifier as either candidate signer certificates or candidate issuer CAs, should the verifier choose to make use of them. These functions do the "adding".

Reply:
Rephrased as follows, taking your suggestion into account:
```
Signer certificates and intermediate CA certificates are usually added
to a CMS structure by L<CMS_sign_ex(3)> and L<CMS_sign(3)>, yet further
such certificates may be added in advance by directly calling these functions.
The verifier of the message can use the included certificates
as candidate signer certificates and for chain building.
These functions may also be used before calling L<CMS_verify(3)> to add
fallback signer certificates or intermediate CA certificates for chain building.
```